src/parser/Lexer.cpp

```diff
@@ -1375,17 +1343,15 @@ PassRefPtr<Scanner::ScannerResult> Scanner::scanStringLiteral()
     }
 
     if (isPlainCase) {
-        StringView str(this->source, start + 1, this->index - 1);
-        auto ret = adoptRef(new (createScannerResult()) Scanner::ScannerResult(this, Token::StringLiteralToken, str, this->lineNumber, this->lineStart, start, this->index, true));
-        ret->octal = octal;
-        ret->plain = true;
-        return ret;
+        StringView* str = new StringView(this->source, start + 1, this->index - 1);
```

Comment:
ASSERT(str != nullptr);

Reply:
`StringView` is allocated by GC. If memory alloc is failed, GC will throw or warn error.